src/node_sqlite.cc

```diff
@@ -585,27 +590,80 @@ bool DatabaseSync::ShouldIgnoreSQLiteError() {
   return ignore_next_sqlite_error_;
 }
 
+std::optional<std::string> ValidateDatabasePath(Environment* env,
```

Comment:
I don't really see the value of this function. SQLite already validates paths.

Reply:
This makes `node:sqlite` compatible with other [modules](https://nodejs.org/api/fs.html#fspromisesaccesspath-mode) that deal with `paths`. It validates the `path` passed in the argument. SQLite itself will perform other kinds of validations.

For example, this will work in the current version of Node

```js
const db = new DatabaseSync('\0my\0path.db'); // SQLite accepts this
```

After this PR, this will throw an error due to the null bytes in the string.

Reply:
Alright, makes sense.